### PR TITLE
Create AvailabilitySet for Azure clusters

### PR DESCRIPTION
**Pull request by mrIncompetent**
**What this PR does / why we need it**:
Create AvailabilitySet for Azure clusters and set it for each machine.

**Release note**:
```release-note cloud provider
Create AvailabilitySet for Azure clusters and set it for each machine
```

### CODE REVIEW

File: api/pkg/provider/cloud/azure/provider.go [hunk 1]
```diff
@@ -533,8 +563,65 @@ func (a *azure) InitializeCloudProvider(cluster *kubermaticv1.Cluster, update pr
...
+		cluster, err = update(cluster.Name, func(updatedCluster *kubermaticv1.Cluster) {
+			updatedCluster.Spec.Cloud.Azure.AvailabilitySet = asName
+			updatedCluster.Finalizers = append(updatedCluster.Finalizers, finalizerAvailabilitySet)
+		})
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	if cluster.Spec.Cloud.Azure.AvailabilitySet == "" {
```
Comment (xrstf) [inline api/pkg/provider/cloud/azure/provider.go]: I know that Azure sucks, but do we really need to create the same AS twice in order for it to work?

Comment (mrIncompetent) [reply api/pkg/provider/cloud/azure/provider.go]: Oh thanks for mentioning - seems i screwed up a merge

Comment (mrIncompetent) [reply api/pkg/provider/cloud/azure/provider.go]: fixed in b97a7a32ddef1f9c923ecb08bdb7bed0a918df73

**Review by kdomanski — APPROVED** [review]

File: api/pkg/provider/cloud/azure/provider.go [hunk 2]
```diff
@@ -204,14 +219,29 @@ func (a *azure) CleanUpCloudProvider(cluster *kubermaticv1.Cluster, update provi
...
-			updatedCluster.Spec.Cloud.Azure.ResourceGroup = ""
 			updatedCluster.Finalizers = kuberneteshelper.RemoveFinalizer(updatedCluster.Finalizers, finalizerResourceGroup)
 		})
 		if err != nil {
 			return nil, err
 		}
 	}
 
+	if kuberneteshelper.HasFinalizer(cluster, finalizerAvailabilitySet) {
+		glog.Infof("cluster %q: deleting availability set %q", cluster.Name, cluster.Spec.Cloud.Azure.AvailabilitySet)
```
Comment (p0lyn0mial) [inline api/pkg/provider/cloud/azure/provider.go:230]: do we want to set a log level for this message ?

File: api/pkg/provider/cloud/azure/provider.go [hunk 3]
```diff
@@ -533,8 +563,48 @@ func (a *azure) InitializeCloudProvider(cluster *kubermaticv1.Cluster, update pr
 		}
 	}
 
+	if cluster.Spec.Cloud.Azure.AvailabilitySet == "" {
+		asName := "cluster-" + cluster.Name
+		glog.Infof("cluster %q: ensuring AvailabilitySet %q", cluster.Name, asName)
```
Comment (p0lyn0mial) [inline api/pkg/provider/cloud/azure/provider.go:568]: I think we should also set a log level here ?

File: api/pkg/provider/cloud/azure/provider.go (hunk 2 above)
Comment (mrIncompetent) [reply api/pkg/provider/cloud/azure/provider.go:230]: We're consitently not doing it in the azure provider, thus i would rather do it in a dedicated PR

File: api/pkg/provider/cloud/azure/provider.go (hunk 3 above)
Comment (mrIncompetent) [reply api/pkg/provider/cloud/azure/provider.go:568]: We're consitently not doing it in the azure provider, thus i would rather do it in a dedicated PR

**Review by p0lyn0mial — APPROVED** [review]

**Pull request merged**